internal/functional/lazyoptional/lazy_optional.go

```
@@ -27,6 +27,24 @@ func Empty[T any]() LazyOptional[T] {
+		return Empty[T]()
+	}
+	return Some[T](*ptr)
+}
+
+// OfError returns a LazyOptional containing the provided error if it is non-nil,
+// or an empty LazyOptional otherwise.
+func OfError(err error) LazyOptional[error] {
+	if err == nil {
+		return Empty[error]()
+	}
+	return Some(err)
```

golangci-lint (2.5.0)

[error] return with no blank line before

(nlreturn)